enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,427 @@
+
+#### Degrade Mechanism
+
+The MSBIC will degrade the worker `MachineConfigPool` via a new [MachineConfigPoolConditionType](https://github.com/openshift/api/blob/master/machineconfiguration/v1/types.go#L492). This would be an API change, but a fairly simple one is it only adding a new condition type. The node controller(another sub controller within the MCC) would then [check for this condition](https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/status.go#L142C34-L142C34) and degrade the worker pool, effectively degrading the operator.
+
+As mentioned in the above section, degrading will only happen when the patching of the MachineSet fails. This is likely due to a temporary API server outage and will resolve itself without user intervention. The degrade condition is calculated at the end of a sync loop. In the case of multiple such failures within a single sync loop, the message for the degrade will be accumulated to include the `MachineSets` associated with all the failures. 
+
+#### Reverting to original bootimage
+
+The proposal will introduce a CR, `MachineSetBootImageHistory` to store the boot image history associated with a given machineset. By providing this CR and accompanying documentation, the user will be able to restore their machinesets to an earlier state if they wish to do so. 
+
+### Workflow Description
+
+It is important to note that there would be two "opt-in" knobs while this feature is under TechPreview. The user would first have to turn on the feature gate, and then the opt-in mechanism. The secondary knob is necessary as some customers may want to keep their boot images static when this feature leaves TechPreview.
+
+See the API extension section for examples of how this feature can be turned on and off. 
+
+#### Variation and form factor considerations [optional]
+
+Any form factor using the MCO and `MachineSets` will be impacted by this proposal. So case by case:
+- Standalone OpenShift: Yes, this is the main target form factor.
+- microshift: No, as it does [not](https://github.com/openshift/microshift/blob/main/docs/contributor/enabled_apis.md) use `MachineSets`.
+- Hypershift: No, Hypershift does not have this issue.
+
+##### Supported platforms
+
+The initial release(phase 0) will support GCP. In future releases, we will add in support for remaining platforms as we gain confidence in the functionality and understand the specific needs of those platforms. For platforms that cannot be supported, we aim to atleast provide documentation to perform the boot image updates manually. Here is an exhaustive list of all the platforms:
+
+- gcp
+- aws
+- azure
+- alibabacloud
+- nutanix
+- powervs
+- openstack
+- vsphere
+- baremetal
+- libvirt
+- ovirt
+- ibmcloud
+
+This work will be tracked in [MCO-793](https://issues.redhat.com/browse/MCO-793).
+
+##### Cluster API backed machinesets
```

Comment:
@damdo @nrb Would you review this section in case there's anything that might break here that I haven't picked up on

Reply:
Finally managed to read this section. I think that the recommendation sounds reasonable and consistent with upstream approaches, but agree that the next pass will need to address MachineDeployments (mentioned [here](https://github.com/openshift/enhancements/pull/1496/files#r1463392787)).

Reply:
Thanks for taking a look! @nrb I have updated the Overview and the Cluster API section to address this. Please let me know if that is sufficient and/or if you have any other concerns!